src/main/java/edu/uw/edm/edmzuulproxy/certificateauthorizer/service/impl/CertificateAuthorizerServiceImpl.java

```diff
@@ -0,0 +1,159 @@
+package edu.uw.edm.edmzuulproxy.certificateauthorizer.service.impl;
+
+import com.google.common.base.Preconditions;
+
+import org.apache.logging.log4j.util.Strings;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import edu.uw.edm.edmzuulproxy.certificateauthorizer.CertificateAuthorizationRepository;
+import edu.uw.edm.edmzuulproxy.certificateauthorizer.model.CertificateAuthorization;
+import edu.uw.edm.edmzuulproxy.certificateauthorizer.service.CertificateAuthorizerService;
+import edu.uw.edm.edmzuulproxy.security.User;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author Maxime Deravet Date: 2019-01-28
+ */
+@Service
+@Slf4j
+public class CertificateAuthorizerServiceImpl implements CertificateAuthorizerService {
+    public static final String HTTP_METHOD_WILDCARD = "*";
+    public static final String UW_GROUPS_WILDCARD = "*";
+    public static final String DEFAULT_SEPARATOR = ",";
+    public static final String UW_GROUPS_SEPARATOR = DEFAULT_SEPARATOR;
+    public static final String HTTP_METHODS_SEPARATOR = DEFAULT_SEPARATOR;
+    public static final String CONTACT_EMAILS_SEPARATOR = DEFAULT_SEPARATOR;
+
+
+    private CertificateAuthorizationRepository certificateAuthorizationRepository;
+
+    @Autowired
+    public CertificateAuthorizerServiceImpl(CertificateAuthorizationRepository certificateAuthorizationRepository) {
+        this.certificateAuthorizationRepository = certificateAuthorizationRepository;
+    }
+
+    @Override
+    @Cacheable(cacheNames = "authorized-uri")
+    public boolean isAllowedForUri(String certificateName, HttpMethod httpMethod, String uri, User user) {
```

Comment:
Maybe we can precompute a mapping of certname -> uriRegex, and then check that the cert has access to the uri, before hitting this method. If the cache is including the uri, maybe it wont be getting hit very often because the uri will include things like contentItemId, etc